services/src/main/java/org/jboss/windup/web/services/rest/WindupEndpointImpl.java

```diff
@@ -55,18 +65,49 @@
     private Queue executorQueue;
 
     /**
+     * To avoid problems with passing detached entities,
+     * load related entities during cloning
+     */
+    protected AnalysisContext cloneExecution(AnalysisContext originalContext)
+    {
+        Set<Package> allPackages = new HashSet<>();
+
+        allPackages.addAll(originalContext.getIncludePackages());
+        allPackages.addAll(originalContext.getExcludePackages());
+
+        List<Long> packageIds = allPackages.stream().map(aPackage -> aPackage.getId())
+                .collect(Collectors.toList());
+
+        AnalysisContext clone = originalContext.clone();
```

Comment:
I'm trying to realize that potential race condition (although I believe extremely rare) when we have multi user scenario (race condition won't happen for single user..) and `AnalysisContext` changes between saving it into DB and calling execute.  (I'm still not convinced if it is really worth to deal with it, because IMHO the probability of that race condition is so extremely small... But @OndraZizka  wants it for robustness).

Well, `AnalysisContext` might not match, but applications, packages and all related entities won't change. So they could be received from persistence context.